kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/GodotPlugin.kt

```diff
@@ -36,10 +45,29 @@ class GodotPlugin : Plugin<Project> {
                 )
             }
 
-            jvm.target.compilations.getByName("main").defaultSourceSet {
-                dependencies {
-                    compileOnly("com.utopia-rise:godot-library:${GodotBuildProperties.godotKotlinVersion}")
+            fun KotlinWithJavaCompilation<KotlinJvmOptions>.configureSourceSets(includeEntrySourceDir: Boolean) {
+                defaultSourceSet {
+                    kotlin.srcDirs("src/main/kotlin")
+                    if (includeEntrySourceDir) {
+                        kotlin.srcDirs(project.buildDir.resolve("godot-entry"))
+                    }
+                    dependencies {
+                        compileOnly("com.utopia-rise:godot-library:${GodotBuildProperties.godotKotlinVersion}")
+                    }
+                }
+            }
+
+            val mainCompilation = jvm.target.compilations.getByName("main").apply {
+                configureSourceSets(true)
+            }
+
+            val dummy = jvm.target.compilations.create("dummy") {
+                defaultSourceSet.dependencies {
+                    main.allDependencies.forEach {
+                        implementation(it)
+                    }
```

Comment:
Here I'm not so sure if this really covers all cases and does not imply some problems.
I tried it by adding an external dependency (see comments at the corresponding places) and it worked.
Maybe @piiertho or @raniejade have a better idea or can assure me that this will work?